### PR TITLE
Improved performance

### PR DESCRIPTION
**Pull request by Jazz-Man**
Slow array function used in loop

### CODE REVIEW

**Comment by davialexandre** [comment]
Interesting. I wonder how much faster is this compared to the previous version. @Jazz-Man could you give some numbers to show the performance difference?

**Comment by colemanw** [comment]
Thanks for the PR. Did you run the unit tests before submitting this @Jazz-Man?

File: Civi/Api4/Service/Schema/Joiner.php [hunk 1]
```diff
@@ -83,11 +83,18 @@ protected function getPath($baseTable, $joinPath) {
...
           $baseTable = $lastLink->getTargetTable();
         }
       }
+      
+      if (PHP_VERSION_ID < 50600) {
+        $fullPath = call_user_func_array('array_merge', $fullPath);
+      }
+      else {
+        $fullPath = array_merge(...$fullPath);
+      }
```
Comment (colemanw) [inline Civi/Api4/Service/Schema/Joiner.php]: This really needs a comment to explain what's going on.

Comment (Jazz-Man) [reply Civi/Api4/Service/Schema/Joiner.php]: Everything is described in the official [documentation](http://php.net/manual/en/functions.arguments.php#functions.variable-arg-list) and  [sample code](http://php.net/manual/ru/functions.arguments.php#121579) (also from official documentation)

Comment (colemanw) [reply Civi/Api4/Service/Schema/Joiner.php]: Since 5.6 is now the minimum recommended php version for CiviCRM, I think this conditional can be removed.

**Pull request closed without merging**